core/templates/dev/head/domain/topic/TopicUpdateService.js

```diff
@@ -218,17 +218,35 @@ oppia.factory('TopicUpdateService', [
           throw Error('A deleted subtopic cannot be restored');
         });
         if (newlyCreated) {
-          var changeList = UndoRedoService.getCommittableChangeList();
-          for (var i = 0; i < changeList.length; i++) {
-            if (changeList[i].subtopic_id === subtopicId) {
-              changeList[i].is_deleted = true;
+          // Get the current change list.
+          var currentChangeList = UndoRedoService.getChangeList();
+          var indicesToDelete = [];
+          for (var i = 0; i < currentChangeList.length; i++) {
+            var backendChangeObject =
```

Comment:
I'm a bit confused about the naming here. By "object" do you mean "dict"? If so I suggest saying "dict", here and elsewhere.

On the other hand, if you mean the domain object, then you don't need the "object" suffix. We don't use that for topic/subtopic/exploration/etc. -- just use the noun itself.